# Rice pan-NLRome
This  GitHub repository archived the code and data generated from the following project:
> **Gao, Shang** ***et al.*** (forthcoming) Constructing rice pan-NLRome and identifying how it was reshaped to response changing pathogenic stress during domestication.

This repository is uploaded by Shang Gao at School of Life Sciences, Tsinghua University.
 
## Visualization code
Codes for figure generation can be found in the following file: 
> `/Figure_Visualization_Code_15OCT2021.ipynb`

R and packages for data manipulation and figure generation are:
> R 3.6.3
> tidyverse
> ggplot2
> gapminder
> socviz
> ggsci
> gridExtra
> stringr
> scales
> ggrepel
> cowplot
> ggsignif
> UpSetR
> ggsankey
> ggmap
> sp
> maptools
> maps

## Data for visualization
The major information table for all NLR genes is:
>`/Data_for_visualization/summary_table_15Oct2021_1043.csv`

This table contians following columns:  
|Column name| Description|
|---|---|
|__gene_ID__| The unique ID for every NLR genes in the rice pan-NLRome.|
|__Assembly_ID__| The ID of every rice genome assembly.
|__Accession_ID__| The ID of every rice genotype.
|__Contig_ID__| The contig where the NLR gene located.
|__Position__| The start position of the NLR gene.
|__Status__| The status of the rice genotype, i.e., cultivar, landrace or wild.
|__Species__| The subspecies information of the genotype.
|__Clustered_Type__| Paired (H2H, "head-to-head"), clustered (Concat), or singleton (-) NLR genes. T2T means the NLR genes are in a "tail-to-tail" arrangement with another NLR gene.
|__Pair_partner_new__| The partner gene of a paired NLR gene.
|__raw_NLR_arch__| The raw NLR architecture, namely, domain configuration, of a NLR generated by Pfam_scan.pl.
|__New2_NLR_domain_info__| The domain information of final NLR architecture optimized with LRRpredictor.
|__New2_NLR_arch__| The final NLR architecture.
|__New2_NLR_class__| The final NLR class.
|__LRR_number__| The number of LRR motif.
|__with_lrr__| Whether a NLR gene comes with LRR domain.
|__HOG__| Hierarchical ortholog group ID.
|__HOG_class__| The class of an HOG.
|__HOG_size__| The number of sequences in an HOG.
|__pangenome_section__| The pan-genome section that an HOG belongs to.


Most figures of this project are generated based on this table.
Other involved data in figure generation are also in `Data_for_visualization` directory.



## Result dataset
The following file includes all identified NLR (without filteration) from the 67 rice accessions:
>`/Result_dataset/rice_panNLRome_AA_Sequence_dataset.tar.gz`

The following file includes are the aligned consensus sequences of NB-ARC domains for all 863 HOGs:
>`/Result_dataset/all_HOG.cons.outgroup_added.mafft.fa`

The following file is the ML tree of the rice pan-NLRome generated by IQ-tree:
>`/Result_dataset/rice_panNLRome_phylogeny_IQ-tree.contree`


## Citation
Gao, Shang et al. (forthcoming) Constructing rice pan-NLRome and identifying how it was reshaped to response changing pathogenic stress during domestication.

